scipy/stats/_pagel.py

```diff
@@ -0,0 +1,466 @@
+    and we pass ``predicted_ranks=[1, 2, 3]``, the ranks we predict for
+    each column of the original table.
+
+    >>> res = pagel(table,                      # data as originally tabulated
+    ...             ranked=False,               # original table is not ranked
+    ...             predicted_ranks=[1, 2, 3],  # we predict increasing rank
+    ...             method="asymptotic"
+    ...             )
+    >>> res
+        method: 'asymptotic'
+        pvalue: 0.0012693433690751756
+     statistic: 133.5
+
+    As presented in [3]_, the *p*-value was calculated based on the asymptotic
+    distribution of the :math:`L` statistic. However, the asymptotic
+    distribution is not very accurate, nor conservative, for :math:`m \leq 12`
+    and :math:`n \leq 8`. Rather than passing in ``method="asymptotic"``,
+    we can allow ``pagel`` to select the appropriate method.
+
+    >>> res = pagel(table, ranked=False, predicted_ranks=[1, 2, 3])
+    >>> res
+        method: 'exact'
+        pvalue: 0.0018191161948127822
+     statistic: 133.5
+
+    Note that ``pagel`` chose to use ``method='exact'`` based on the dimensions
+    of the table and the recommendations in Page's original paper [1]_.
+    This exact *p*-value is greater than that given by the
+    asymptotic approach, but we can still reject the null hypothesis in favor
+    of our alternative at the 99% confidence level.
+    """
+
+    # Possible values of the method parameter and the corresponding function
+    # used to evaluate the p value
+    method = method.lower()
+    methods = {"asymptotic": _l_p_asymptotic,
+               "exact": _l_p_exact,
+               "auto": None}
+    if method not in methods:
+        raise ValueError(f"`method` must be in {set(methods)}")
+
+    ranks = np.array(data, copy=False)
+    if ranks.ndim != 2:  # TODO: relax this to accept 3d arrays?
+        raise ValueError(f"`data` must be a 2d array.")
+
+    m, n = ranks.shape
+    if m < 2 or n < 3:
+        raise ValueError("Page's L is only appropriate for data with two "
+                         "or more rows and three or more columns.")
+
+    if np.any(np.isnan(data)):
+        raise ValueError("`data` contains NaNs, which cannot be ranked "
+                         "meaningfully")
+
+    # ensure NumPy array and rank the data if it's not already ranked
+    if ranked:
+        # Only a basic check on whether data is ranked. Checking that the data
+        # is properly ranked could take as much time as ranking it.
+        if not (ranks.min() >= 1 and ranks.max() <= ranks.shape[1]):
+            raise ValueError("`data` is not properly ranked. Rank the data or "
+                             "pass `ranked=False`.")
+    else:
+        ranks = scipy.stats.rankdata(data, axis=-1)
+
+    # generate predicted ranks if not provided, ensure valid NumPy array
+    if predicted_ranks is None:
+        predicted_ranks = np.arange(n, 0, -1)
```

Comment:
I think I did it this way because from the original paper:

![image](https://user-images.githubusercontent.com/6570539/101702984-d4564280-3a36-11eb-9a4e-b36e9a98e350.png)

Does that make sense?

I'll change it though, if in your experience everyone lays out their tables the other way. I'll have to make a lot of changes to the documentation and probably tests, so we'll have to check carefully that everything is updated.

Reply:
Also:
![image](https://user-images.githubusercontent.com/6570539/101703157-28612700-3a37-11eb-8670-9f2f255234ca.png)

Reply:
I took another look at Odeh's paper, which is the basis for the computation here.  From page 50 of the paper: 

        This paper is concerned with a rank test for testing the null hypothesis

                        H₀:  t₁ = ... = tk  (= 0)

    against the ordered alternative

                        Hₐ: t₁ ≤ ... ≤ tk ,

    where at least one inequality is strict.

And later there is a parenthetical comment "(Here the smallest observation receved the rank 1)".

So the Odeh formulation is based on assigning ranks like `rankdata` does, and the default `predicted_ranks` should be `[1, 2, ..., k]`.

Reply:
Please check the most recent commit - the math and all the doc changes; I think I reversed everything.